crates/polars-plan/src/plans/mod.rs

```diff
@@ -76,15 +69,15 @@ pub enum DslPlan {
     /// Cache the input at this point in the LP
     Cache { input: Arc<DslPlan>, id: usize },
     Scan {
-        sources: Arc<Mutex<DslScanSources>>,
-        // Option as this is mostly materialized on the IR phase.
-        // During conversion we update the value in the DSL as well
-        // This is to cater to use cases where parts of a `LazyFrame`
-        // are used as base of different queries in a loop. That way
-        // the expensive schema resolving is cached.
-        file_info: Arc<RwLock<Option<FileInfo>>>,
+        sources: ScanSources,
+        /// Materialized at IR except for AnonymousScan.
+        file_info: Option<FileInfo>,
         file_options: FileScanOptions,
         scan_type: FileScan,
+        /// Local use cases often repeatedly collect the same `LazyFrame` (e.g. in interactive notebook use-cases),
+        /// so we cache the IR conversion here, as the path expansion can be quite slow (especially for cloud paths).
+        #[cfg_attr(feature = "serde", serde(skip))]
+        cached_ir: Arc<Mutex<Option<IR>>>,
```

Comment:
We had mutexes on multiple fields on the `DSL::Scan`, I just cache the IR here instead of adding another mutex